### PR TITLE
Update fTPM and Authvar drivers

### PR DESCRIPTION
**Pull request by dmcilvaney**
- Removed obsolete services (Checksum, RNG, OPM)
- Update TPM service to be inline with TCG2 spec instead of obsolete TrEE spec
- Update variable driver to use better error codes
- Update UUIDs for both authvar and ftpm TAs

Note: This pull request needs to be in sync with related pull requests updating the fTPM and Authvars TAs. The UUIDs have changed and the driver will no longer be able to communicate with the previous versions of the TAs.

### CODE REVIEW

File: driver/TrEE/TrEE/OpteeVariableService.h [hunk 1]
```diff
@@ -23,12 +23,10 @@ UEFI Variable Services definitions for accessing operations.
 typedef enum _VARIABLE_SERVICE_OPS
 {
     VSGetOp = 0,
+    VSGetNextVarOp,
```
Comment (hbeberman) [inline driver/TrEE/TrEE/OpteeVariableService.h:26]: Why have these rearranged?

File: driver/TrEE/TrEE/VariableService.c [hunk 2]
```diff
@@ -653,17 +653,24 @@ Return Value:
         &TeecOperation,
         &ErrorOrigin);
 
+    *ResultSize = (UINT32)TeecOperation.params[2].value.a;
+    *AuthVarStatus = (UINT32)TeecOperation.params[2].value.b;
+
```
Comment (hbeberman) [inline driver/TrEE/TrEE/VariableService.c:658]: Why did this move?

**Review by hbeberman — APPROVED** [review]

**Comment by hbeberman** [comment]
Is this intended to be merged into public_preview yet or do you want it in a staging branch for TA improvements?

File: driver/TrEE/TrEE/OpteeVariableService.h (hunk 1 above)
Comment (dmcilvaney) [reply driver/TrEE/TrEE/OpteeVariableService.h:26]: They are in the same order as the UEFI spec now (plus exit boot services).

File: driver/TrEE/TrEE/VariableService.c (hunk 2 above)
Comment (dmcilvaney) [reply driver/TrEE/TrEE/VariableService.c:658]: The return value was expected to contain the size even during a failure (specifically EFI_BUFFER_TOO_SMALL). This allows the caller to reallocate a larger buffer to contain the result and try again.

**Comment by dmcilvaney** [comment]
Now targeting update_TAs branch.

File: driver/TrEE/TrEE/FtpmService.c [hunk 3]
```diff
@@ -879,8 +720,8 @@ Return Value:
 
 {
     ULONG OperationStatus;
-    UINT8 PpiFlags;
-    TREE_PPI_VARIABLE PpiValue;
+	UINT32 PpiFlags;
```
Comment (neilsh-msft) [inline driver/TrEE/TrEE/FtpmService.c]: Indentation is off?

File: driver/TrEE/TrEE/OpteeFtpmService.h [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+#define TCG2_BIOS_TPM_MANAGEMENT_FLAG_PP_REQUIRED_FOR_CLEAR 0x02
+
+//
+// From UEFI TcgPhysicalPresence.h
+//
+
+//
+// TCG PP definition for physical presence ACPI function
+//
+#define TCG_ACPI_FUNCTION_GET_PHYSICAL_PRESENCE_INTERFACE_VERSION		1
```
Comment (neilsh-msft) [inline driver/TrEE/TrEE/OpteeFtpmService.h]: Indentation is off for all the constants.

File: driver/TrEE/TrEE/OpteeFtpmService.h [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+#define PPI_SUMBIT_OPERATION2_BLOCKED           3
+
+//
+// Return codes for TCG_ACPI_FUNCTION_GET_USER_CONFIRMATION_STATUS_FOR_REQUEST
+//
+#define PPI_USER_CONFIRMATION_NOT_IMPLEMENTED   0
+#define PPI_USER_CONFIRMATION_FIRMWARE          1
+#define PPI_USER_CONFIRMATION_BLOCKED           2
+#define PPI_USER_CONFIRMATION_PP_REQUIRED       3
+#define PPI_USER_CONFIRMATION_PP_NOT_REQUIRED   4
```
Comment (neilsh-msft) [inline driver/TrEE/TrEE/OpteeFtpmService.h]: Missing final newline in file.

File: driver/TrEE/TrEE/OpteeFtpmService.h (hunk 4 above)
Comment (dmcilvaney) [reply driver/TrEE/TrEE/OpteeFtpmService.h]: VS added tabs to some lines, fixed.

File: driver/TrEE/TrEE/FtpmService.c (hunk 3 above)
Comment (dmcilvaney) [reply driver/TrEE/TrEE/FtpmService.c]: VS added tabs to some lines, fixed.

File: driver/TrEE/TrEE/OpteeFtpmService.h (hunk 5 above)
Comment (dmcilvaney) [reply driver/TrEE/TrEE/OpteeFtpmService.h]: fixed

**Review by christopherco — APPROVED** [review]

**Pull request merged**